pmoves/services/hi-rag-gateway-v2/routes/geometry.py

```
@@ -0,0 +1,574 @@
+                "conf": p.get("conf"),
+                "meta": p.get("meta"),
+                "ref_id": p.get("ref_id") or p.get("doc_id") or p.get("video_id"),
+            })
+    pts.sort(key=lambda x: (x.get("conf") or 0.0, x.get("proj") or 0.0), reverse=True)
+    top_pts = pts[:k]
+
+    if mode == "learned":
+        try:
+            from transformers import pipeline
+        except (ImportError, ModuleNotFoundError) as exc:
+            raise HTTPException(500, f"transformers not available: {type(exc).__name__}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)